app/src/common/shared/org/mozilla/vrbrowser/browser/components/GeckoEngineSession.kt

```diff
@@ -0,0 +1,40 @@
+package org.mozilla.vrbrowser.browser.components
+
+import mozilla.components.concept.engine.EngineSession
+import mozilla.components.concept.engine.EngineSessionState
+import mozilla.components.concept.engine.Settings
+import org.json.JSONObject
+import org.mozilla.geckoview.GeckoSession
+import org.mozilla.vrbrowser.browser.engine.SessionStore
+
+class GeckoEngineSession(
+        val geckoSession: GeckoSession
+): EngineSession() {
+    constructor() :
+        this(SessionStore.get().createSession(false, false).geckoSession)
```

Comment:
Is this empty constructor, that explicitly creates a session, needed? Looks like that would cause problems because we never want to create one and always use the one you already created in FxR code?

Reply:
It seems that in some cases the an extension needs to create a new tab and uses that constructor to do so?

https://github.com/mozilla-mobile/android-components/blob/d642612386e8d9880ff83a78737e065b8ac870c7/components/browser/engine-gecko-nightly/src/main/java/mozilla/components/browser/engine/gecko/GeckoEngine.kt#L80-L82

        override fun onToggleActionPopup(extension: WebExtension, action: Action): EngineSession? {
            return webExtensionDelegate?.onToggleActionPopup(extension, GeckoEngineSession(runtime), action)
        }

Reply:
Oh, I see! Then I guess this makes sense if that will then let you know about this tab too. :)

Reply:
Yes, calling `SessionStore.get().createSession` should both add the session to the SessionStore sessions list and to the BrowserStore.